pykechain/client.py

```diff
@@ -1378,6 +1378,60 @@ def create_proxy_model(
         )
         return self._create_part(action='create_proxy_model', data=data, **kwargs)
 
+    def _create_parts_bulk(
+            self,
+            parts: List[Dict],
+            **kwargs
+    ) -> PartSet:
+        """
+        Create multiple part instances simultaneously.
+
+        :param parts: list of dicts, each specifying a part instance. Available fields per dict:
+            * name: str
+            * parent_id: str of parent instance
+            * model_id: str of part model
+            * properties: list of dicts, each specifying a property to update. Available fields per dict:
+                * name: str
+                * value: Any
+                * model_id: str of property model
+        :param kwargs:
+        :return: list of Part instances
+        :rtype list
+        """
+        check_list_of_dicts(
+            parts,
+            'parts',
+            [
+                'name',
+                'parent_id',
+                'model_id',
+                'properties',
+            ],
+        )
+        for part in parts:
+            check_list_of_dicts(
+                part.get('properties'),
+                'properties',
+                [
+                    'name',
+                    'value',
+                    'model_id',
+                ],
+            )
+
+        parts = {"parts": parts}
+        # prepare url query parameters
+        query_params = kwargs
+        query_params.update(API_EXTRA_PARAMS['parts2'])
+        response = self._request('POST', self._build_url('parts2_bulk_create'),
+                                 params=query_params, json=parts)
+        if response.status_code != requests.codes.created:
```

Comment:
can also receive a 202 ACCEPTED in case we do a async call.

You can make the code ready for this when you look at the bulk clone activity command. There a 'asynchronous' flag determines if the call is done async and the 202 is then checked. I suggest copy/paste that code here for future proof-ness (if you catch my drift, not yet in the backend api, but will come).

Reply:
Good feedback, understand, will implement.

Reply:
see `pykechain/client.py:1197` in the branch `clone_activity_api_renaming`

```python
        params['async_mode'] = asynchronous

        response = self._request('POST', self._build_url('activities_bulk_clone'),
                                 json=data, params=params)

        if (asynchronous and response.status_code != requests.codes.accepted) or \
                (not asynchronous and response.status_code != requests.codes.created):  # pragma: no cover
            raise APIError("Could not clone Activities.", response=response)

        cloned_activities = [Activity2(d, client=self) for d in response.json()['results']]

        if isinstance(activity_parent, Activity2):
            activity_parent._populate_cached_children(cloned_activities)

        return cloned_activities
```

Reply:
added, please re-review